src/lib.rs

```diff
@@ -1316,9 +1316,36 @@ impl CecConnectionCfg {
             return Err(CecConnectionResultError::LibInitFailed);
         }
 
-        if unsafe { libcec_open(connection.1, port.as_ptr(), open_timeout) } == 0 {
-            return Err(CecConnectionResultError::AdapterOpenFailed);
-        }
+        let open_timeout = connection.0.open_timeout.as_millis() as u32;
+        match &connection.0.port {
+            Some(port) => {
+                let port = CString::new(port.as_str()).expect("Invalid port name");
+                if unsafe { libcec_open(connection.1, port.as_ptr(), open_timeout) } == 0 {
+                    return Err(CecConnectionResultError::AdapterOpenFailed);
+                }
+            }
+            None => {
+                let mut adapters = MaybeUninit::<[libcec_sys::cec_adapter_descriptor; 1]>::uninit();
+                let num_adapters = unsafe {
+                    libcec_sys::libcec_detect_adapters(
+                        connection.1,
+                        adapters.as_mut_ptr() as *mut _,
+                        1,
+                        std::ptr::null(),
+                        true as i32,
+                    )
+                };
+
+                if num_adapters == 0 {
+                    return Err(CecConnectionResultError::NoAdapterFound);
+                }
+
+                let port = unsafe { adapters.assume_init() }[0].strComName;
+                if unsafe { libcec_open(connection.1, port.as_ptr(), open_timeout) == 0 } {
```

Comment:
I decided to call `libcec_open` in both branches of this match block to the avoid the overhead of converting `strCommName` to a CString.